src/compiler/crystal/semantic/type_lookup.cr

```diff
@@ -250,7 +250,7 @@ class Crystal::Type
 
         # Check the case of T resolving to a number
         if type_var.is_a?(Path)
-          type = @root.lookup_path(type_var)
+          type = lookup_type_var(type_var)
```

Comment:
This might work fine regarding aliases if you do this:

```suggestion
          type = in_generic_args { lookup_type_var(type_var) }
```

Reply:
To be honest, I can't remember why this is needed, but that's how it's done everywhere in this method when a type is looked up.